### PR TITLE
Expose MaxConnsPerNode from AeroSpike to allow easier performance tuning

### PR DESCRIPTION
**Pull request by munan-geo**
## Description

This PR is to expose MaxConnsPerNode from AeroSpike to allow easier performance tuning.

Note 1: We leave `MaxConnsPerNode` as 100 which is the default from Aerospike. However, Aerospike has also updated other default settings, see the [change reasons](https://discuss.aerospike.com/t/client-1772-client-configurations-changes-reasons/9699) from Aerospike forum. We will update AsyncMaxCommands to 100 when we bump up Aerospike version so we align with Aerospike settings. Also, the relationship between asyncMaxConnsPerNode and MaxConnsPerNode can be found from [here](https://aerospike.com/apidocs/csharp/html/f_aerospike_client_asyncclientpolicy_asyncmaxconnspernode).
Note 2: We are running into code cov issue and it seems that we didn't have token setup properly for this repo. I was instructed to comment out the code cov temporarily.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair-incubator/AeroSharp/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

**Review by amber-ashraf — APPROVED** [review]
rubber stamp - would need another reviewer to review it before merging

**Review by brittany-alves — APPROVED** [review]

**Review by draco2003 — APPROVED** [review]
LGTM

File: src/AeroSharp/Connection/ClientProvider.cs
Comment (farmerau) [inline src/AeroSharp/Connection/ClientProvider.cs:24]: Since this is an async policy, we should consider setting asyncMaxConnsPerNode (https://aerospike.com/apidocs/csharp/html/f_aerospike_client_asyncclientpolicy_asyncmaxconnspernode) as noted in the docs for maxConnsPerNode https://aerospike.com/apidocs/csharp/html/f_aerospike_client_clientpolicy_maxconnspernode

Because it's an async policy, it's not clear to me that setting maxConnsPerNode would have much effect.

**Review by farmerau — CHANGES_REQUESTED** [review]

File: src/AeroSharp/Connection/ClientProvider.cs
Comment (munan-geo) [reply src/AeroSharp/Connection/ClientProvider.cs:24]: Thank you Austin for reviewing this PR. My response below
* It is true that this is creating an `AsyncClientPolicy` object here, but since `AsyncClientPolicy` inherits from `ClientPolicy`, I would think it is legal to setup a property from parent class when creating a child object like the other properties `user` / `password`?
* We didn't update `asyncMaxConnsPerNode`, and let it default to -1 and use `maxConnsPerNode` according to [here](https://aerospike.com/apidocs/csharp/html/f_aerospike_client_asyncclientpolicy_asyncmaxconnspernode). If we set the default value to 100, then we are changing Aerospike library default behavior (check aerospike source code [here](https://github.com/aerospike/aerospike-client-csharp/blob/master/AerospikeClient/Async/AsyncClientPolicy.cs#L104)) and I would prefer not to do so as a wrapper.
* We have been playing with `maxConnsPerNode` for a while within different teams, and it did impact the behavior a lot.

Please let me know your thought!

Comment (farmerau) [reply src/AeroSharp/Connection/ClientProvider.cs:24]: Ah, I see.

Quoting the relevant portion of that doc here:

> If the value is -1, the value will be set to [maxConnsPerNode](https://aerospike.com/apidocs/csharp/html/f_aerospike_client_clientpolicy_maxconnspernode).

Thanks for following up on this.

**Review by farmerau — APPROVED** [review]
LGTM!

**Pull request merged**
